iot-hub/Samples/device/ConventionBasedOperations/readme.md

```diff
@@ -0,0 +1,563 @@
+#### Using convention-aware API (new):
+
+```csharp
+// Retrieve the client's properties.
+ ClientProperties properties = await _deviceClient.GetClientPropertiesAsync(cancellationToken);
+
+// To fetch the value of client reported property "serialNumber".
+bool isSerialNumberReported = properties.TryGetValue("serialNumber", out string serialNumberReported);
+
+
+// To fetch the value of service requested "targetTemperature" value.
+bool isTargetTemperatureUpdateRequested = properties.Writable.TryGetValue("targetTemperature", out double targetTemperatureUpdateRequest);
+```
+
+### Retrive component-level client properties:
+
+#### Using non-convention-aware API (old):
+
+```csharp
+// Retrieve the client's properties.
+Twin properties = await _deviceClient.GetTwinAsync(cancellationToken);
+
+// To fetch the value of client reported property "serialNumber" under component "thermostat1".
+JToken serialNumberJToken = null;
+bool isSerialNumberReported = properties.Properties.Reported.Contains("thermostat1")
+    && ((JObject)properties.Properties.Reported["thermostat1"]).TryGetValue("serialNumber", out serialNumberJToken);
+
+if (isSerialNumberReported)
+{
+    string serialNumberReported = serialNumberJToken?.ToObject<string>();
+}
+
+// To fetch the value of service requested "targetTemperature" value under component "thermostat1".
+JToken targetTemperatureUpdateRequestJToken = null;
+bool isTargetTemperatureUpdateRequested = properties.Properties.Desired.Contains("thermostat1")
+    && ((JObject)properties.Properties.Desired["thermostat1"]).TryGetValue("targetTemperature", out targetTemperatureUpdateRequestJToken);
+
+if (isTargetTemperatureUpdateRequested)
+{
+    double targetTemperatureUpdateRequest = (double)(targetTemperatureUpdateRequestJToken?.ToObject<double>());
+}
+```
+
+#### Using convention-aware API (new):
+
+```csharp
+// Retrieve the client's properties.
+ ClientProperties properties = await _deviceClient.GetClientPropertiesAsync(cancellationToken);
+
+// To fetch the value of client reported property "serialNumber" under component "thermostat1".
+bool isSerialNumberReported = properties.TryGetValue("thermostat1", "serialNumber", out string serialNumberReported);
+
+
+// To fetch the value of service requested "targetTemperature" value under component "thermostat1".
+bool isTargetTemperatureUpdateRequested = properties.Writable.TryGetValue("thermostat1", "targetTemperature", out double targetTemperatureUpdateRequest);
+```
+
+### Update top-level property:
+
+#### Using non-convention-aware API (old):
+
+```csharp
+// Update the property "serialNumber".
+var propertiesToBeUpdated = new TwinCollection
+{
+    ["serialNumber"] = "SR-1234",
+};
+await _deviceClient.UpdateReportedPropertiesAsync(propertiesToBeUpdated, cancellationToken);
+```
+
+#### Using convention-aware API (new):
+
+```csharp
+// Update the property "serialNumber".
+var propertiesToBeUpdated = new ClientPropertyCollection
+{
+    ["serialNumber"] = "SR-1234",
+};
```

Comment:
Do we want this to be part of the samples? It doesn't seem clear when we'd use `AddRootProperty` versus this.

Reply:
are you suggesting we use the explicit method call instead of an initializer list?